.github/workflows/ci.yaml

```diff
@@ -26,38 +29,85 @@ jobs:
   metadata:
     runs-on: ubuntu-22.04
     outputs:
-      TRIAL_BRANCH: ${{ steps.meta.outputs.TRIAL_BRANCH }}
+      BUILD_DATE: ${{ steps.date.outputs.BUILD_DATE }}
+      PUBLISH: ${{ steps.if-publish.outputs.PUBLISH }}
+      BUMP_MANIFEST: ${{ steps.if-bump-manifest.outputs.BUMP_MANIFEST }}
     steps:
-      - name: Set optional trial branch
-        id: meta
+      - name: Set build date
+        id: date
+        shell: bash -x -e {0}
+        run: |
+          BUILD_DATE=$(TZ='US/Los_Angeles' date '+%Y-%m-%d')
+          echo "BUILD_DATE=${BUILD_DATE}" >> $GITHUB_OUTPUT
+
+      - name: Determine whether results will be 'published'
+        id: if-publish
+        shell: bash -x -e {0}
+        run: |
+          echo "PUBLISH=${{ github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && inputs.PUBLISH) }}" >> $GITHUB_OUTPUT
+
+      - name: Determine whether need to bump manifest 
+        id: if-bump-manifest
         shell: bash -x -e {0}
         run: |
-          if [[ -z "${{ inputs. TRIAL_BRANCH}}" ]]; then
-            echo "TRIAL_BRANCH=${{ github.sha }}"
-          else
-            echo "TRIAL_BRANCH=${TRIAL_BRANCH}"
-          fi | tee $GITHUB_OUTPUT
+          echo "BUMP_MANIFEST=${{ github.event_name == 'schedule' }}" >> $GITHUB_OUTPUT
 
   amd64:
     needs: metadata
     uses: ./.github/workflows/_ci.yaml
     with:
       ARCHITECTURE: amd64
-      TRIAL_BRANCH: ${{ needs.metadata.outputs.TRIAL_BRANCH }}
+      BUILD_DATE: ${{ needs.metadata.outputs.BUILD_DATE}}
+      BUMP_MANIFEST: ${{ needs.metadata.outputs.BUMP_MANIFEST}}
     secrets: inherit
  
   arm64:
     needs: metadata
     uses: ./.github/workflows/_ci.yaml
     with:
       ARCHITECTURE: arm64
-      TRIAL_BRANCH: ${{ needs.metadata.outputs.TRIAL_BRANCH }}
+      BUILD_DATE: ${{ needs.metadata.outputs.BUILD_DATE}}
+      BUMP_MANIFEST: ${{ needs.metadata.outputs.BUMP_MANIFEST}}      
     secrets: inherit
 
+  publish:
+    runs-on: ubuntu-22.04
+    needs: [metadata, amd64, arm64]
+    if: ${{ !cancelled() }} && needs.metadata.outputs.PUBLISH
```

Comment:
`${{...}}` is not required but OK in a `if:` key (this is a GitHub Actions peculiarity IMHO)
Either way, it looks a bit strange since we have half of the condition inside of `${{...}}` and the other half out of it.

Reply:
Removed ${{}} around canceled()